src/array.rs

```diff
@@ -31,7 +31,7 @@ pub trait ArrayEncoding: Encoding {
 #[cfg_attr(docsrs, doc(cfg(feature = "generic-array")))]
 pub trait ArrayDecoding {
     /// Big integer which decodes a `GenericArray`.
-    type Output: Integer;
+    type Output: ArrayEncoding + Integer + Rem<NonZero<Self::Output>, Output = Self::Output>;
```

Comment:
Bounding on `ArrayEncoding` is fine, but something like `Rem<NonZero<Self::Output>, Output = Self::Output>` should probably get captured at a higher level than here.

Reply:
Like `Integer`?